src/app_unit.rs

```diff
@@ -289,6 +289,92 @@ impl Au {
     }
 }
 
+#[cfg(feature = "num_traits")]
+impl num_traits::ToPrimitive for Au {
+    // Float conversions
+
+    #[inline]
+    fn to_f32(&self) -> Option<f32> {
+        Some(self.to_f32_px())
+    }
+
+    #[inline]
+    fn to_f64(&self) -> Option<f64> {
+        Some(self.to_f64_px())
+    }
+
+    // Signed int conversions
+
+    #[inline]
+    fn to_i8(&self) -> Option<i8> {
+        None
```

Comment:
Not much familiar with this, but shouldn't it be `self.to_px().to_i8()` or such?
Analogous for all returning `None`.

Reply:
`None` basically means "we don't support this cast", so it seemed like a conservative default for:

- Unsigned integers. As casting an `Au` to an unsigned integer is likely a logic error.
- Signed integers that have a size smaller than the range that `Au` supports. Again, we probably don't want to do this?

Taking a look at the [implementations](https://docs.rs/num-traits/latest/src/num_traits/cast.rs.html#288) that `num_traits` provides for analogous stdlib types, it looks like it does runtime checks to see whether the value is in a range that can be converted.

So we could do that if you think that's better?

Reply:
Even if it may be a weird conversion, if the value can be represented, it seems to make more sense to try to cast. But I will defer to Martin.